### PR TITLE
only return visible elements as focusable

### LINKED ISSUE #14317: A11y_AzureDataStudioSQL_Project_Database_CreateProjectFromDatabase_Keyboard: Keyboard focus is moving outside of the "Create Project From Database" pane after hitting shift + tab key.
“[Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

**GitHubTags:** #A11yMAS;#A11yTCS;#A11ySev2;#SH_AzureDataStudioSQL_Feb2021;#SQL Azure Data Studio;#Win10;#FTP;#DesktopApp;#WCAG2.4.3;#Keyboard;#FTP;#AINotInScope;

### Environment Details:
Application Name: Azure Data Studio
Windows Version: Win10

### Repro Steps:

1. Launch the application-Azure Database Project and install extension SQL Database Projects.
2. Tab till SQL Database Project Home Screen with heading "Projects".
3. Tab to connections select your server and connect to it.
4. Few databases will appear under your connection then press shift+f1 on any database
5. Tab till "Create project from database" and hit enter.
6. Tab to the "Select connection" button under "Source database" title and press shift + tab key.
7. Check if focus is moving outside of the pane

### Actual:
Keyboard focus is moving outside of the "Create Project From Database" pane after hitting shift + tab key.

Same issue is observed in below screen:

1. data tier application

### Expected:
Keyboard focus should not go outside of the "Create Project From Database" pane after hitting shift + tab key.

### User Impact:
Keyboard only users will face difficulty in navigating again in the pane.

### LINKED ISSUE #14318: A11y_AzureDataStudioSQL_Project_CreateProjectFromDatabase_Keyboard: Keyboard focus is getting trapped as it is not moving by hitting tab key when keyboard focus is on "cancel" button
“[Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

**GitHubTags:** #A11yMAS;#A11yTCS;#A11ySev2;#SH_AzureDataStudioSQL_Feb2021;#SQL Azure Data Studio;#Win10;#FTP;#DesktopApp;#WCAG2.1.2;#Keyboard;#AINotInScope;

### Environment Details:
Application Name: Azure Data Studio
Windows Version: Windows10

### Repro Steps:

1. Launch the application-Azure Database Project and install extension SQL Database Projects.
2. Tab till SQL Database Project Home Screen with heading "Projects".
3. Tab to connections select your server and connect to it.
4. Few databases will appear under your connection then press shift+f10 on any database
5. Tab till "Create project from database" and hit enter.
6. Tab to the "Cancel" button and press tab key.
7. Check if focus is moving to next focusable control or not.

### Actual:
Keyboard focus is getting trapped as it is not moving to the next focusable control which is "Select connection" button by hitting tab key when keyboard focus is on "cancel" button.

### Expected:
Keyboard focus should go to the next focusable control which is "Select connection" button by hitting tab key

### User Impact:
Keyboard only users will face difficulty in navigating inside the pane.

### LINKED ISSUE #14519: A11y_AzureDataStudioSQL_Project_Database_CreateProjectFromDatabase_Keyboard: Keyboard focus is moving outside of the "Create Project From Database" pane after hitting shift + tab key.
“[Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

**GitHubTags**: #A11yMAS;#A11yTCS;#A11ySev2;#SH_AzureDataStudioSQL-MAC-Feb2021;#SQL Azure Data Studio;#MAC;#FTP;#DesktopApp;#WCAG2.4.3;#Keyboard;#AINotInScope;

### Environment Details:
Application Name: Azure Data Studio
Mac OS Version: Big Sur Version 11.2.2

### Repro Steps:

1. Launch the application-Azure Database Project and install extension SQL Database Projects.
2. Tab till SQL Database Project Home Screen with heading "Projects".
3. Tab to connections select your server and connect to it.
4. Few databases will appear under your connection then right click on any database
5. Tab till "Create project from database" and hit enter.
6. Tab to the "Select connection" button under "Source database" title and press shift + tab key.
7. Check if focus is moving outside of the pane.

### Actual:
Keyboard focus is moving outside of the "Create Project From Database" pane after hitting shift + tab key.

Same issue is observed in below screen:
Data tier application

### Expected:
Keyboard focus should not go outside of the "Create Project From Database" pane after hitting shift + tab key.

### User Impact:
Keyboard only users will face difficulty in navigating again in the pane.

### LINKED ISSUE #14520: A11y_AzureDataStudioSQL_Project_CreateProjectFromDatabase_Keyboard: Keyboard focus is getting trapped as it is not moving by hitting tab key when keyboard focus is on "cancel" button.
“[Check out Accessibility Insights! ](https://accessibilityinsights.io/)- Identify accessibility bugs before check-in and make bug fixing faster and easier.”

**GitHubTags**: #A11yMAS;#A11yTCS;#A11ySev2;#SH_AzureDataStudioSQL-MAC-Feb2021;#SQL Azure Data Studio;#MAC;#FTP;#DesktopApp;#WCAG2.1.2;#Keyboard;#AINotInScope;

### Environment Details:
Application Name: Azure Data Studio
Mac OS Version: Big Sur Version 11.2.2

### Repro Steps:

1. Launch the application-Azure Database Project and install extension SQL Database Projects.
2. Tab till SQL Database Project Home Screen with heading "Projects".
3. Tab to connections select your server and connect to it.
4. Few databases will appear under your connection then right click on any database
5. Tab till "Create project from database" and hit enter.
6. Tab to the "Cancel" button and press tab key.
7. Check if focus is moving to next focusable control or not.

### Actual:
Keyboard focus is getting trapped as it is not moving to the next focusable control which is "Select connection" button by hitting tab key when keyboard focus is on "cancel" button.

### Expected:
Keyboard focus should go to the next focusable control which is "Select connection" button by hitting tab key

### User Impact:
Keyboard only users will face difficulty in navigating inside the pane.

### PR DESCRIPTION
**Pull request by alanrenmsft**
This PR fixes #14519
This PR fixes #14520 
This PR fixes #14318
This PR fixes #14317

the elements returned by the query contains invisible elements and caused these 2 issues. the fix is to further filter the list to only return visible elements.

also I am moving this function out so that it can be reused (I will reuse it for the query result filter dialog).

### CODE REVIEW

File: src/sql/workbench/browser/modal/modal.ts [hunk 1]
```diff
@@ -2,30 +2,32 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
-import 'vs/css!./media/modal';
```
Comment (alanrenmsft) [inline src/sql/workbench/browser/modal/modal.ts:5]: reorgnized the imports, the only actual change is added import entry for the new method I added.

File: src/sql/base/browser/dom.ts [hunk 2]
```diff
@@ -44,3 +44,23 @@ export function convertSizeToNumber(size: number | string | undefined): number {
...
+ * @returns The focusable elements
+ */
+export function getFocusableElements(container: HTMLElement): HTMLElement[] {
+	const elements = [];
+	container.querySelectorAll(tabbableElementsQuerySelector).forEach((element: HTMLElement) => {
+		const style = window.getComputedStyle(element);
+		// We should only return the elements that are visible. There are many ways to hide an element, for example setting the
+		// visibility attribute to hidden/collapse, setting the display property to none, or if one of its ancestors is invisible.
+		if (element.offsetWidth > 0 && element.offsetHeight > 0 && style.visibility === 'visible') {
+			elements.push(element);
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/dom.ts:62]: It makes more sense to just use filter for this : 

```
const elements = container.querySelectorAll(tabbableElementsQuerySelector).filter(element => {
    const style = window.getComputedStyle(element);
    return element.offsetWidth > 0 && element.offsetHeight > 0 && style.visibility === 'visible';
}
return elements;
```

**Review by Charles-Gagnon — APPROVED** [review]

File: src/sql/base/browser/dom.ts (hunk 2 above)
Comment (alanrenmsft) [reply src/sql/base/browser/dom.ts:62]: the return value of the querySelectorAll() is not an array, I have to convert it to array first in order to use filter method.

**Comment by udeeshagautam** [comment]
seeing this late... thanks @alanrenmsft  for this :)

**Pull request merged**